javaHapiValidatorLambda/src/main/java/software/amazon/fwoa/Handler.java

```diff
@@ -38,4 +56,49 @@ public ValidatorResponse handleRequest(String event, Context context) {
         return validate;
 
     }
+
+    private List<String> getBucketObjects(String bucketName) {
+        try {
+            List<String> objectKeys = new ArrayList<String>();
+            ListObjectsV2Request req = new ListObjectsV2Request().withBucketName(bucketName);
+            ListObjectsV2Result result;
+            do {
+                result = s3.listObjectsV2(req);
+
+                for (S3ObjectSummary objectSummary : result.getObjectSummaries()) {
+                    objectKeys.add(objectSummary.getKey());
+                }
+                // If there are more than maxKeys keys in the bucket, get a continuation token
+                // and list the next objects.
+                String token = result.getNextContinuationToken();
+                req.setContinuationToken(token);
+            } while (result.isTruncated());
+            log.info("found " + objectKeys.size() + " keys");
+            return objectKeys;
+        } catch (Exception e) {
+            throw new Error(e);
+        }
+    }
+
+    private DownloadedGuidesHolder downloadObjects(List<String> keys, String bucketName) {
+        // keep track of .index.json objects
+        List<IGObject> indices = new ArrayList<IGObject>();
+        List<IGObject> resources = new ArrayList<IGObject>();
+        for (String key : keys) {
+            try (S3ObjectInputStream s3Object = s3.getObject(bucketName, key).getObjectContent()) {
+                IGObject bucketObj = new Utils.IGObject(key, IOUtils.toString(s3Object));
```

Comment:
Can we build this `toString` functionality into the `IGObject` class? So `IGObject` can have a `getContentObject` and `getContentString` method. This way it gives more flexibility to downstream logic to choose from what to use.